### DISCUSSION TITLE
Suggestion for possible fix of macOS caching issue

### DISCUSSION THREAD

**Discussion by maximilian-kn** [hbldh/bleak#1429] [General]
In the past few days, I had severe problems trying to connect to a Raspberry Pi via bleak from my MacBook.
I did a fresh install of Raspbian and changed the hostname, which resulted in unexpected horrors. MacOS did not update the hostname and was still picking up the old hostname. Deleting Bluetooth cache files and rebooting did not fix the issue either.

Today, I figured that when pairing the MacBook and the Pi via the Pi's `bluetoothctl` its hostname gets somehow updated and bleak finds the Pi under its actual hostname.
I just left the Pi as a known device in the Mac's Bluetooth settings, and now everything is working perfectly.

I am not sure about any reproduceability of this, but it might be worth mentioning this as an unofficial way of tackling macOS caching issues as [Troubleshooting](https://bleak.readthedocs.io/en/latest/troubleshooting.html?highlight=mac%20os%20cache#handling-os-caching-of-ble-device-services) has no fixes for macOS yet.

Bleak version: 0.21.1
macOS: Ventura 13.6
Raspbian: Raspberry Pi OS (64-bit), Kernel version: 6.1, Debian version: 11 (bullseye)
Bluetoothctl version: 5.55

**Comment by dlech** [comment]
I'm not sure I fully understand the issue here. Is Bleak running on the Mac or on the Pi?

**Reply by maximilian-kn** [reply]
Bleak is running on Mac and I was trying to connect to the Pi to run pytests to check several steps of an initial enrollment via Bluetooth.

**Reply by dlech** [reply]
In my experience, this seems like quite standard behavior for all OSes and both Bluetooth Classic and BLE when a device is paired and the name changes. It seems like it always takes a while for the OS to decide to update the name or you can unpair and repair to update the name right away.
